### PR TITLE
Directly Fix #2839

### LINKED ISSUE #2839: Adding files that are symlinks directly
Unix commands when you reference a symlink directly resolve it. IPFS does not.

Example:
```bash
mkdir /tmp/hello
echo "World" > /tmp/hello/world
ln -s /tmp/hello /tmp/sym
cd /tmp/sym
ipfs add -r .

cp -r . ../sym1
ls -l .. | grep sym
```

`cp` resolved the symlink and copied whole directory, IPFS added an meaningless symlink.

### PR DESCRIPTION
**Pull request by kevina**
Revert the original fix for #2839 and instead fix the original issue so that adding the current directory works as expected without changing the semantics of how symbolic links are handled.

### CODE REVIEW

File: test/sharness/t0044-add-symlink.sh [hunk 1]
```diff
@@ -27,34 +23,27 @@ test_add_symlinks() {
...
+		ipfs add -q files/bad > badlink_out
 	'
 
 	test_expect_success "output looks good" '
 		echo "QmWYN8SEXCgNT2PSjB6BnxAx6NJQtazWoBkTRH9GRfPFFQ" > badlink_exp &&
 		test_cmp badlink_exp badlink_out
 	'
-
-	test_expect_success "adding with symlink in middle of path is same as\
-adding with no symlink" '
```
Comment (whyrusleeping) [inline test/sharness/t0044-add-symlink.sh]: This test should still be valid.

Comment (kevina) [inline test/sharness/t0044-add-symlink.sh]: Done.

**Comment by whyrusleeping** [comment]
Alright, This LGTM. @jbenet since this is in response to your review, want to give it a quick look over?

**Comment by jbenet** [comment]
LGTM

**Comment by whyrusleeping** [comment]
Thanks @kevina!

**Comment by NeoTheFox** [comment]
So, just to be clear, would we be able to make `ipfs add -r ` to resolve symlinks if needed now?

**Pull request merged**